### ISSUE TITLE
Low log level makes form-data requests hang indefinitely

### ISSUE THREAD

**Issue by KnorpelSenf** [tdlib/telegram-bot-api#178]
## Description

When the Bot API server is run locally, some requests hang. In particular, requests that need to be sent with multipart/form-data are affected, such as `sendDocument` and `sendPhoto`. The request is parsed, but no photo message is created. The connection to the bot remains open.

I have spend a good deal of time trying to figure out what the relevant factors are. I tried a number of combinations.

- Instance hosted on api.telegram.org and local instance
- Commit 05b3d5f and commit bcd0c8e
- Log level 5 and log level 15
- Debian 10 and Ubuntu 20.04
- Using `--local` and not using `--local`
- Plain HTTP on localhost, and going through a local reverse proxy with self-signed certificate
- [Node.js](https://nodejs.org/) and [Deno](https://deno.land/)
- [grammY](https://grammy.dev) and [Telegraf](https://telegraf.js.org)

These are the results of my testing.

- Only occurs on local instances.
- Occurs on both investigated commits.
- Only occurs with a low log level. It may be between 5 and 15, but the higher the log level, the less often a request hangs.
- Occurs on both Debian and Ubuntu.
- The `--local` flag has no effect on this issue.
- Only occurs over plain HTTP.
- Only occurs on Node.js.
- Only occurs using grammY.

This means that you will not see the issue if any of the above constraints is not given.
For example, using grammY over plain HTTP locally on Deno will not reproduce the issue.

## Possible cause

I can only speculate that this is a concurrency issue. My reasoning would be that
- SSL is slower than no SSL,
- remote is slower than local,
- huge amounts of log output are slower than lo logs,
- Telegraf is slower than grammY,
- Deno is a very new runtime and might still be slower than Node.

All of these things would prevent the issue from happening. But again, this is just a guess.

## Reproduction steps

Make sure that `/tmp/td` does not exist yet. Now run the built Bot API server locally.

```bash
mkdir -p /tmp/td/{dir,temp-dir}
./bin/telegram-bot-api --local --api-id=XXX --api-hash=xxxxx --dir=/tmp/td/dir --temp-dir=/tmp/td/temp-dir -v 5
```

Next, create a grammY bot project in any fresh directory on the same machine. You need to have Node.js installed. grammY needs version `^12.20.0 || >=14.13.1` of Node.

```sh
npm install grammy --save
```

Create a file called `bot.js` with the following code.

```js
const { Bot, InputFile } = require('grammy')

const bot = new Bot(process.env.BOT_TOKEN, {
    client: {
        apiRoot: 'http://localhost:8081',
        baseFetchConfig: { compress: true, agent: undefined },
    },
})

const fileData = new TextEncoder().encode('abc')

// Log all incoming updates
bot.use(async (ctx, next) => {
    console.log(ctx.update)
    await next()
})

bot.command('start', ctx => ctx.reply('Up and running!'))
bot.command('race', async (ctx) => {
    // Calls `sendDocument` under the hood
    const message = await ctx.replyWithDocument(new InputFile(fileData));
    console.log(message);
});

bot.start({ drop_pending_updates: true })
```

Set the token of your bot obtained from [@BotFather](https://telegram.me/botfather) as env variable, and run the bot.

```sh
export BOT_TOKEN=123:abc-d
export DEBUG='grammy*' # optionally view debug output

node bot.js
```

In a Telegram client, send `/start` to your bot to check if it is running.

Next, send `/race` a couple of times. Sooner or later, no file will be sent back, and the bot hangs.

By default, grammY processes all updates sequentially. This makes your code predictable. In this case, it will cause your bot to hang until the `sendDocument` call completes (which is never does).

**Comment by levlam** [comment]
Thanks. This was fixed in TDLib and will be fixed in the local Bot API server with the next update.

Only multipart/form-data requests sent using chunked Transfer-Encoding were affected. And only if the sender is slow enough to send the chunks one-by-one.

**Comment by levlam** [comment]
Fixed in Bot API 5.3.2.
